frontend/src/components/dialog-pro/index.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+    <el-dialog
+        :title="title"
+        v-model="dialogVisible"
+        :destroy-on-close="true"
+        :close-on-click-modal="false"
+        :width="size"
+    >
+        <div v-if="slots.content">
+            <slot name="content"></slot>
+        </div>
+        <el-row v-else>
+            <el-col :span="22" :offset="1">
+                <slot></slot>
+            </el-col>
+        </el-row>
+
+        <template #footer v-if="slots.footer">
+            <slot name="footer"></slot>
+        </template>
+    </el-dialog>
+</template>
+
+<script lang="ts" setup>
+import { computed, useSlots } from 'vue';
+defineOptions({ name: 'DrawerPro' });
+
+const props = defineProps({
+    title: String,
+    size: {
+        type: String,
+        default: 'normal',
+    },
+    modelValue: {
+        type: Boolean,
+        default: false,
+    },
+});
+
+const slots = useSlots();
+
+const emit = defineEmits(['update:modelValue', 'close']);
+
+const size = computed(() => {
+    switch (props.size) {
+        case 'small':
+            return '30%';
+        case 'normal':
+            return '40%';
+        case 'large':
+            return '50%';
+        case 'full':
+            return '100%';
+        case '60%':
+            return '60%';
+        default:
+            return '50%';
+    }
+});
+
+const dialogVisible = computed({
+    get() {
+        return props.modelValue;
+    },
+    set(value: boolean) {
+        emit('update:modelValue', value);
+    },
+});
+</script>
```

Comment:
Based on the provided snippet, here are some observations and suggested improvements:

1. **Size Property Naming**: `size` property is redundant since it's already stored as an instance variable named `size`.
2. **Slots Implementation**: The `<template>` tags do not match their `<slot>` directives, specifically because `<template>` elements should be placed inside the `<div>` element for inline slot processing instead of placing them outside `<slot>s`.
3. **Dialog Width Management**: Use `$attrs` to access the width directly when creating this component rather than setting via `.width="size"`.

Here's the refined code based on these points:

```html
<div class="drawer-pro" @click="$emit('close')">
    <div :class="'drawer-' ~ $attrs.width || "normal"' ref="drawer"> <!-- Adjusted line -->
      <!-- Content goes here -->
    </div> 
</div>
```

This version addresses most concerns listed above while maintaining a consistent syntax across all components within this package if it were included laterally, like in `el-container`.